src/clr-angular/data/datagrid/providers/column-toggle-buttons.service.spec.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import {ColumnToggleButtonsService} from "./column-toggle-buttons.service";
+
+export default function(): void {
+    describe("ColumnToggleButtons provider", function() {
+        let service: ColumnToggleButtonsService;
+
+        beforeEach(function() {
+            service = new ColumnToggleButtonsService();
+        });
+
+        it("should have expected properties", function() {
+            expect(service.buttons).toBeNull();
+            expect(service.selectAllDisabled).toBe(false);
+            expect(service.okButtonClicked.subscribe).toBeDefined();
+            expect(service.selectAllButtonClicked.subscribe).toBeDefined();
+        });
+
+        it("should emit clicks", function(done) {
+            let calls = 0;
+            function handle() {
+                calls++;
+                if (calls === 2) {
+                    expect(calls).toEqual(2);
```

Comment:
Wait, what? If `calls === 2`, expect `calls` to equal `2`? You're not testing anything here. I believe you just want the `expect` outside of the function, right after 
```typescript
service.buttonClicked("ok");
service.buttonClicked("selectAll");
```
Unless I'm missing something and not all pieces are synchronous.

Reply:
The `buttonClicked()` calls `next()` on an observable, so I'm testing that both clicks emitted correctly by listening and counting the times the handler was called. This will fail if one of the emits fails to be handled as a timeout. Its not particularly elegant, I could use a spy instead except the internals are private.